source/AI.cpp

```diff
@@ -579,21 +578,26 @@ void AI::Step(Command &activeCommands)
 	bool opportunisticEscorts = !Preferences::Has("Turrets focus fire");
 	bool fightersRetreat = Preferences::Has("Damaged fighters retreat");
 	const int npcMaxMiningTime = GameData::GetGamerules().NPCMaxMiningTime();
-	for(const auto &it : ships)
+
+	// Copy the ship pointers for parallel execution
+	vector<shared_ptr<Ship>> shipVector(ships.begin(), ships.end());
+	for_each(execution::par, shipVector.begin(), shipVector.end(), [&](auto &it)
```

Comment:
```suggestion
	for_each(execution::par_unseq, shipVector.begin(), shipVector.end(), [&](auto &it)
```
Would this be even faster?
I am reading into this stuff currently, so just asking of interest.

Reply:
It cannot be unsequential because it acquires locks (in Dictionary, for instance).